contracts/extension/plugin/Map.sol

```diff
@@ -3,61 +3,103 @@ pragma solidity ^0.8.0;
 
 import "../interface/plugin/IMap.sol";
 
-import "../Multicall.sol";
 import "../../openzeppelin-presets/utils/EnumerableSet.sol";
 
 /**
  *  TODO:
  *      - Remove OZ EnumerableSet external dependency.
  */
 
-contract Map is IMap, Multicall {
+abstract contract Map is IMap {
     using EnumerableSet for EnumerableSet.Bytes32Set;
 
     EnumerableSet.Bytes32Set private allSelectors;
 
-    mapping(address => EnumerableSet.Bytes32Set) private selectorsForExtension;
-    mapping(bytes4 => address) private extension;
+    mapping(address => EnumerableSet.Bytes32Set) private selectorsForPlugin;
+    mapping(bytes4 => Plugin) private pluginForSelector;
 
-    constructor(ExtensionMap[] memory _extensionsToRegister) {
-        uint256 len = _extensionsToRegister.length;
+    constructor(Plugin[] memory _pluginsToRegister) {
+        uint256 len = _pluginsToRegister.length;
         for (uint256 i = 0; i < len; i += 1) {
-            _setExtensions(_extensionsToRegister[i].selector, _extensionsToRegister[i].extension);
+            _setPlugin(_pluginsToRegister[i]);
         }
     }
 
-    function _setExtensions(bytes4 _selector, address _extension) internal {
-        require(allSelectors.add(bytes32(_selector)), "REGISTERED");
+    function setPlugin(Plugin memory _plugin) external {
+        require(_canSetPlugin(), "Not authorized");
 
-        extension[_selector] = _extension;
-        selectorsForExtension[_extension].add(bytes32(_selector));
+        _setPlugin(_plugin);
+    }
+
+    function updatePlugin(Plugin memory _plugin) external {
+        require(_canSetPlugin(), "Not authorized");
+
+        _updatePlugin(_plugin);
+    }
+
+    function removePlugin(bytes4 _selector) external {
+        require(_canSetPlugin(), "Not authorized");
+
+        _removePlugin(_selector);
+    }
+
+    function _setPlugin(Plugin memory _plugin) internal {
+        require(allSelectors.add(bytes32(_plugin.selector)), "REGISTERED");
```

Comment:
Now that we're not facing the contract size limit, let's clean up require statement error messages.

Let's follow the rule that error messages begin as "{Contract Name}: ..." so that it is easier to locate where errors are occurring. The error messages themselves don't need to be elaborate.